hrm-domain/hrm-service/migrations/20240329215610-sanitize-identifiers-remove-blank-spaces.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  up: async queryInterface => {
+    await queryInterface.sequelize.query(`
+    DO $$
+      DECLARE idx public."Identifiers";
+    BEGIN
+      -- For each identifier in Identifier with spaces
+      FOR idx IN (SELECT * FROM "Identifiers" WHERE "identifier" LIKE '% %') LOOP
+        -- Remove any blank spaces that the identifiers might have
+        UPDATE "Identifiers" SET "identifier" = replace(idx."identifier", ' ', '') WHERE id = idx.id;
+      END LOOP;
```

Comment:
I was not expecting a for loop in here 😅 
I was going to ask if this couldn't be done without a loop, but it doesn't really matter in the end.

Reply:
Haha Postgres loops goes brrrr. I also don't love it like this but is the way I felt safer, and since is a single time migration probably won't have a negative effect in the performance anyways :P

Reply:
Eh, migrations are the place for dirty dirty SQL like this, just don't ever let me see it in SQL the service runs! 😂😂😂 

Out of interest though, why did you do it this way? Seems like a simple UPDATE statement would have been easier to write? Is it so it will partially complete if it hits an error with any records rather than fully rolling back? That's the only practical difference I can see?

Reply:
Is 
`UPDATE "Identifiers" SET "identifier" = replace("identifier", ' ', '') WHERE  "identifier" LIKE '% %';`
supposed to do the same thing? I wasn't sure if `replace` would work in this context, but didn't tried to be honest. May be good to change now if that's allowed :P

Reply:
As far as I can see, yes that's the same. Or even `UPDATE "Identifiers" SET "identifier" = replace("identifier", ' ', '')` :-P

Reply:
I suppose that might spam audit logs though, so do your way